docs/TUTORIAL.md

```diff
@@ -1,3 +1,305 @@
 # Tutorial
 
-Coming soon!
+The **ksonnet** [readme][readme] 
+shows you how to create a default 
+`deployment.json` file that lets you deploy an nginx container 
+to an existing Kubernetes cluster. This tutorial shows you how to:
+
+* Modify the deployment using the built-in **ksonnet** mixins
+* Write a custom mixin library for Fluentd 
+* Add the custom Fluentd mixin to your application deployemtn
+
+## Prerequisites
+
+This tutorial assumes that you have performed the following 
+tasks. For details, see the [readme][readme].
+
+* Installed **Jsonnet**
+* Cloned the **ksonnet** repository locally
+* Installed and configured the VisualStudio Code extension 
+(optional)
+* Created a test Kubernetes cluster
+
+NOTE: All import paths are relative to the root of the 
+*ksonnet** repository.
+
+## Modify the default deployment
+
+**ksonnet** lets you configure or modify any Kubernetes object. In the following 
+exercises, you create an explicit IngressController and Service from the nginx 
+deployment described in the [readme]. You simplify the 
+process of defining and adding these new objects by taking advantage of 
+**ksonnet**'s built-in mixins.
+
+First, define base configuration values, dependent objects, and the IngressController:
+
+```javascript
```

Comment:
I've sometimes seen people link to Github gists---do we want to do this, or assume that people will simply copy + paste? I'm ok either way.